### PR TITLE
Document how to use the kind tool and the Tekton/plumbing convenience script

### PR DESCRIPTION
**Pull request by mrutkows**
# Changes

This PR is a promised follow-on to PR https://github.com/tektoncd/pipeline/pull/3955 (suggested by Andrea and Scott) which intends to document the use of the `kind` tool and specifically use of the convenience script(s) provided in the Tekton plumbing repo. 

It addresses the reference issue https://github.com/tektoncd/pipeline/issues/3970 which captures the request more fully.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [X] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [X] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed (N/A)
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)

/kind documentation

# Release Notes

```release-note
Documenting how to use the kind tool to setup the development environment.
```

### CODE REVIEW

**Comment by mrutkows** [comment]
As I started looking at the inner workings of the script and providing a small "delete" example, I realized more was going on with the script and decided to create https://github.com/tektoncd/plumbing/pull/849 in the plumbing repo. where the script resides to host canonical information that could be referenced in DEVELOPMENT.md to avoid too much duplication.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -265,7 +266,7 @@ The recommended minimum development configuration is:
 
 <!-- TODO: Someone needs to validate the cluster-version-->
 1. Create a GKE cluster (with `--cluster-version=latest` but you can use any
-    version 1.18 or later):
```
Comment (vdemeester) [inline DEVELOPMENT.md:268]: Any reason to revert that back ? on `main` (and most likely on 0.24 too), we have a minimum version of k8s at 1.18.

Comment (mrutkows) [reply DEVELOPMENT.md:268]: Not sure how I reverted this... will put back.

Comment (mrutkows) [reply DEVELOPMENT.md]: Done ;)

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -381,12 +381,69 @@ kubectl set env deployments --all SYSTEM_NAMESPACE=${TARGET_NAMESPACE} -n ${TARG
...
 - Builds and push images with the new namespace to your container registry and
 - Update all Tekton Objects in K8s using these images
 
 It will also update the default system namespace used for K8s `deployments` to the new value for all subsequent `kubectl` commands.
 
+---
+
+### Standing up a K8s cluster with Tekton using the `kind` tool
+
+An alternative to standing up your own K8s cluster and installing Tekton using `ko` is by using the [kind](https://kind.sigs.k8s.io/) tool. It was designed to help create and run local,  Kubernetes clusters in Docker to assist in local development and testing.
```
Comment (sbwsg) [inline DEVELOPMENT.md:395]: nit: looks like there might be a word missing here?  `create and run local,  Kubernetes clusters`

**Review by sbwsg — APPROVED** [review]

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -98,7 +99,7 @@ To [build, deploy and run your Tekton Objects with `ko`](#install-pipeline), you
 
 1. `GOROOT`: Set `GOROOT` to the location of the Go installation you want `ko` to use for builds.
 
-    > **NOTE** You may need to set `GOROOT` if you installed Go tools to a a non-default location or have multiple Go versions installed.
+    > **Note**: You may need to set `GOROOT` if you installed Go tools to a a non-default location or have multiple Go versions installed.
```
Comment (pritidesai) [inline DEVELOPMENT.md:102]: NIT: extra `a` here 🙃  `to a a non-default location`

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -381,12 +381,69 @@ kubectl set env deployments --all SYSTEM_NAMESPACE=${TARGET_NAMESPACE} -n ${TARG
...
+- `Docker`: `kind` also requires Docker to be installed and running locally. Use the [Get Docker](https://docs.docker.com/get-docker/) instructions.
+
+#### Create a cluster with Tekton components
+
+Change into the Tekton plumbing repository you cloned and invoke the script:
+
+```shell
+cd plumbing
+./hack/tekton_in_kind.sh
+```
```
Comment (pritidesai) [inline DEVELOPMENT.md:414]: I haven't tried these instructions myself but excited to experiment with it. Thank you @mrutkows for documenting them in the pipeline repo.

**Comment by pritidesai** [comment]
thanks @mrutkows I have updated the release notes 

/lgtm

File: DEVELOPMENT.md (hunk 2 above)
Comment (mrutkows) [reply DEVELOPMENT.md:395]: I guess that I did not actually need a comma there between local and K8s. Can remove...

File: DEVELOPMENT.md (hunk 3 above)
Comment (mrutkows) [reply DEVELOPMENT.md]: Fixed the "a a" that was from previous PR ;)

File: DEVELOPMENT.md (hunk 2 above)
Comment (mrutkows) [reply DEVELOPMENT.md]: @sbwsg Fixed

**Comment by pritidesai** [comment]
thanks again @mrutkows 

/lgtm

**Pull request merged**